src/sentry/sentry_metrics/querying/visitors.py

```diff
@@ -47,6 +47,9 @@ class QueryConditionVisitor(ABC, Generic[TVisited]):
     """
 
     def visit_group(self, condition_group: ConditionGroup) -> ConditionGroup:
+        if not condition_group:
```

Comment:
Small efficiency improvement. This is done assuming that visitors exclusively mutate expressions that have at least one element.